From e144df5b0bfce27e7eac0a76d9736ec920c1d993 Mon Sep 17 00:00:00 2001
From: Dinh Nguyen <dinguyen@opensource.altera.com>
Date: Thu, 9 Oct 2014 17:54:51 -0500
Subject: [PATCH 008/151] FogBugz #236669: Add a Kconfig for ILC driver

commit  538d37166b6d95e1bce1679ce961a6a371a2b13b from
https://github.com/altera-opensource/linux-socfpga.git
branch is socfpga-5.4.64-lts

Without a Kconfig option, there's really no way to build the driver.

Signed-off-by: Dinh Nguyen <dinguyen@opensource.altera.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
Integrated-by: Jun Zhang <jun.zhang@windriver.com>
---
 drivers/misc/Kconfig  | 6 ++++++
 drivers/misc/Makefile | 1 +
 2 files changed, 7 insertions(+)

diff --git a/drivers/misc/Kconfig b/drivers/misc/Kconfig
index d0f82cbd7a44..2f86062ba600 100644
--- a/drivers/misc/Kconfig
+++ b/drivers/misc/Kconfig
@@ -504,6 +504,12 @@ config ALTERA_SYSID
 	help
 	This enables Altera System ID soft core driver.
 
+config ALTERA_ILC
+	tristate "Altera Interrupt Latency Counter driver"
+	help
+	  This enables the Interrupt Latency Counter driver for the Altera
+	  SOCFPGA platform.
+
 source "drivers/misc/c2port/Kconfig"
 source "drivers/misc/eeprom/Kconfig"
 source "drivers/misc/cb710/Kconfig"
diff --git a/drivers/misc/Makefile b/drivers/misc/Makefile
index 73386d903283..25c647caae93 100644
--- a/drivers/misc/Makefile
+++ b/drivers/misc/Makefile
@@ -45,6 +45,7 @@ obj-y				+= lis3lv02d/
 obj-$(CONFIG_USB_SWITCH_FSA9480) += fsa9480.o
 obj-$(CONFIG_ALTERA_STAPL)	+=altera-stapl/
 obj-$(CONFIG_ALTERA_HWMUTEX)	+= altera_hwmutex.o
+obj-$(CONFIG_ALTERA_ILC)	+= altera_ilc.o
 obj-$(CONFIG_ALTERA_SYSID)	+= altera_sysid.o
 obj-$(CONFIG_INTEL_MEI)		+= mei/
 obj-$(CONFIG_VMWARE_VMCI)	+= vmw_vmci/
-- 
2.26.1

